crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -173,16 +173,40 @@ impl HostFnBuilder {
 }
 
 type ContextMap = Lazy<Arc<Mutex<HashMap<u64, Arc<CallContext>>>>>;
+/// Map from an instance to it's context for a call.
 static CONTEXT_MAP: ContextMap = Lazy::new(Default::default);
 
-type InstanceCache = HashMap<u64, Arc<Mutex<Instance>>>;
-type ZomeKey = [u8; 32];
-type DnaZomeKey = (DnaHash, ZomeKey);
-type DnaZomeCache = Lazy<Arc<Mutex<HashMap<DnaZomeKey, InstanceCache>>>>;
-static INSTANCE_CACHE: DnaZomeCache = Lazy::new(Default::default);
 static CONTEXT_KEY: AtomicU64 = AtomicU64::new(0);
 
-const INSTANCE_CACHE_SIZE: usize = 20;
+/// Create a key for the instance cache.
+/// It will be [WasmHash..DnaHash..context_key] all as bytes.
+fn instance_cache_key(wasm_hash: &WasmHash, dna_hash: &DnaHash, context_key: u64) -> [u8; 32] {
+    let mut bits = [0u8; 32];
+    for (i, byte) in wasm_hash
+        .get_raw_32()
+        .iter()
+        .zip(dna_hash.get_raw_32().iter())
+        .map(|(a, b)| a ^ b)
+        .take(24)
+        .enumerate()
+    {
+        bits[i] = byte;
+    }
+    for (i, byte) in (24..32).zip(context_key.to_be_bytes()) {
```

Comment:
90% of processors are little endian, using LE is then a noop

```suggestion
    for (i, byte) in (24..32).zip(context_key.to_le_bytes()) {
```